step-functions-data-science-sdk/step_functions_mlworkflow_processing/step_functions_mlworkflow_scikit_learn_data_processing_and_model_evaluation.ipynb

```diff
@@ -0,0 +1,987 @@
+   "source": [
+    "Before introducing the script you use for data cleaning, pre-processing, and feature engineering, inspect the first 20 rows of the dataset. The target is predicting the `income` category. The features from the dataset you select are `age`, `education`, `major industry code`, `class of worker`, `num persons worked for employer`, `capital gains`, `capital losses`, and `dividends from stocks`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd\n",
+    "\n",
+    "input_data = \"s3://sagemaker-sample-data-{}/processing/census/census-income.csv\".format(\n",
+    "    region\n",
+    ")\n",
+    "df = pd.read_csv(input_data, nrows=10)\n",
+    "df.head(n=10)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To run the scikit-learn preprocessing script as a processing job, create a `SKLearnProcessor`, which lets you run scripts inside of processing jobs using the scikit-learn image provided."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sklearn_processor = SKLearnProcessor(\n",
+    "    framework_version=\"0.20.0\",\n",
+    "    role=role,\n",
+    "    instance_type=\"ml.m5.xlarge\",\n",
+    "    instance_count=1,\n",
+    "    max_runtime_in_seconds=1200,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook cell writes a file `preprocessing.py`, which contains the pre-processing script. You can update the script, and rerun this cell to overwrite `preprocessing.py`. You run this as a processing job in the next cell. In this script, you\n",
+    "\n",
+    "* Remove duplicates and rows with conflicting data\n",
+    "* transform the target `income` column into a column containing two labels.\n",
+    "* transform the `age` and `num persons worked for employer` numerical columns into categorical features by binning them\n",
+    "* scale the continuous `capital gains`, `capital losses`, and `dividends from stocks` so they're suitable for training\n",
+    "* encode the `education`, `major industry code`, `class of worker` so they're suitable for training\n",
+    "* split the data into training and test datasets, and saves the training features and labels and test features and labels.\n",
+    "\n",
+    "Our training script will use the pre-processed training features and labels to train a model, and our model evaluation script will use the trained model and pre-processed test features and labels to evaluate the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%writefile preprocessing.py\n",
+    "\n",
+    "import argparse\n",
+    "import os\n",
+    "import warnings\n",
+    "\n",
+    "import pandas as pd\n",
+    "import numpy as np\n",
```

Comment:
`isort`:

```suggestion
    "import numpy as np\n",
    "import pandas as pd\n",
```

Reply:
Will be happy to incorporate the changes and resolve them after the initial merge so that we can unblock the launch.